### ISSUE TITLE
Pressing up or down after switching tabs jumps down in the file

### ISSUE THREAD

**Issue by llamicron** [Microsoft/vscode#28016]
- VSCode Version: 1.12.2 (1.12.2)
- OS Version: macOS Sierra 10.12.4

Steps to Reproduce:

1. Have at least one fold active
2. Switch tabs with `ctrl-tab` or `shift-cmd-]`
3. Press down or up arrow

When performing the steps above, _sometimes_ (about 60% of the time) the cursor moves down in the page, at seemingly random distances. Sometimes it moves to the bottom of the page, sometimes just a few lines. This seems to only happen when having a block of code folded. I'm using the Atom layout, and working in Ruby. Is there another shortcut bound to those key presses? I don't think that that's likely as using `ctrl-tab` and `shift-cmd-]` has the same effect.

**Comment by kieferrm** [comment]
@llamicron could you attach an animated gif that shows the behavior? In happens in Ruby files and in what other file types?

**Comment by llamicron** [comment]
I can't reproduce this is any other file type. During the gif, I am switching tabs with `shift-cmd-]` and `shift-cmd-[`, then pressing up on the keyboard. Sometimes the cursor goes own one line (even when pressing up), or 3 lines, or 10. I though that maybe it was traveling to the next fold, but that doesn't seem to be the case. Pretty confusing. 

Here is the contents of `~/.vscode/extensions` if that helps:
```
IBM.output-colorizer-0.1.1
PKief.material-icon-theme-2.0.8
alefragnani.project-manager-0.18.0
andischerer.theme-atom-one-dark-0.0.1
arcticicestudio.nord-visual-studio-code-0.2.0
g3ortega.crystal-0.1.9
ms-vscode.atom-keybindings-2.0.1
rebornix.ruby-0.12.1
slevesque.vscode-multiclip-0.1.2
```

Gif: https://gfycat.com/DisfiguredGiantGermanwirehairedpointer

Part of the video was cut off, but it's more of the same behavior.

**Comment by PunchyRascal** [comment]
Same here:
![tab-bar-focus gif](https://user-images.githubusercontent.com/821272/26880105-a5bc0706-4b93-11e7-925a-2448fde48220.gif)

**Comment by alexandrudima** [comment]
We have fixed this in 1.13.

The cursor view position was not validated after the folding information was restored from the view state. Duplicate of #26251
